turbinia/api/cli/turbinia_client/helpers/formatter.py

```diff
@@ -313,10 +314,132 @@ def generate_markdown(self) -> str:
+          report.append('')
+          continue
+        for task_id, task_attributes in tasks.items():
+          report.append(
+              self.bullet(f'{task_id} - {task_attributes["task_name"]}'))
+          for attribute_name, attribute_value in task_attributes.items():
+            if attribute_name != 'task_name':
+              formatted_name = attribute_name.replace('_', ' ').title()
+              report.append(
+                  self.bullet(f'{formatted_name}: {attribute_value}', level=2))
+        report.append('')
+
+    return '\n'.join(report)
+
+
+class StatsMarkdownReport(MarkdownReportComponent):
+  """A markdown report of the task statistics."""
+
+  def __init__(self, statistics: dict):
+    super().__init__()
+    self._statistics: dict = statistics
+    self.table_dict = {
+        'TASK': [],
+        'COUNT': [],
+        'MIN': [],
+        'MEAN': [],
+        'MAX': []
+    }
+
+  def stat_to_row(self, task: str, stat_dict: dict):
+    """Generates a row of the statistics table.
+    
+    Args:
+      task (str): Name of the current task.
+      stat_dict (dict): Dictionary with information about current row.
+    """
+    self.table_dict['TASK'].append(f'{task}')
+    self.table_dict['COUNT'].append(stat_dict.get('count', ''))
+    self.table_dict['MIN'].append(stat_dict.get('min', ''))
+    self.table_dict['MEAN'].append(stat_dict.get('mean', ''))
+    self.table_dict['MAX'].append(stat_dict.get('max', ''))
+
+  def generate_data_frame(self, markdown: bool = False) -> pandas.DataFrame:
+    """Generates a pandas DataFrame of the statistics table
+
+    Args:
+      markdown (bool): Bool defining if the tasks should be in markdown format.
+    
+    Returns: 
+      data_frame (DataFrame): Statistics table in pandas DataFrame format.
+    """
+    for stat_group, stat_dict in self._statistics.items():
+      stat_group = stat_group.replace('_', ' ').title()
+      if stat_group in ('All Tasks', 'Successful Tasks', 'Failed Tasks',
+                        'Requests'):
+        first_column = self.heading2(
+            f'{stat_group}:') if markdown else stat_group
+        self.stat_to_row(first_column, stat_dict)
+        continue
+      if markdown:
+        self.stat_to_row(self.heading2(f'{stat_group}:'), {})
+      for description, inner_dict in stat_dict.items():
+        first_column = self.bullet(
+            f'{description}:'
+        ) if markdown else f'{stat_group.split(" ")[-1]} {description}'
+        self.stat_to_row(first_column, inner_dict)
+
+    return pandas.DataFrame(self.table_dict)
+
+  def ljust(self, s):
+    """Left justifies cells in data_frame."""
+    s = s.astype(str).str.strip()
+    return s.str.ljust(s.str.len().max())
+
+  def generate_markdown(self) -> str:
+    """Generates a Markdown version of task statistics.
+    
+    Returns:
+      markdown(str): Markdown version of task statistics.
+    """
+    report = [self.heading1('Execution time statistics for Turbinia:')]
+
```

Comment:
We could remove these few empty lines that seem unnecessary

Reply:
Thanks, I removed those here